lib/puppet/provider/foreman_smartproxy/rest_v3.rb

```diff
@@ -19,14 +29,34 @@ def exists?
 
   def create
     post_data = {:smart_proxy => {:name => resource[:name], :url => resource[:url]}}.to_json
-    r = request(:post, 'api/v2/smart_proxies', {}, post_data)
-    raise Puppet::Error.new("Proxy #{resource[:name]} cannot be registered: #{error_message(r)}") unless success?(r)
+    path = 'api/v2/smart_proxies'
+    r = request(:post, path, {}, post_data)
+
+    unless success?(r)
+      error_string = <<~ERROR.strip
+        Error making POST request to #{request_uri(path)}
```

Comment:
Is this really adding anything for the user? IMHO it certainly shouldn't be the first thing. First thing should be the action that failed (the line below this). Then technical details.

Reply:
Agreed. I'll change the ordering in my next round of changes.